File: www/cgi-bin/hello.php
<?php
header('Content-Type: text/html; charset=UTF-8');
echo "<!DOCTYPE html>\n";
echo "<html lang=\"en\">\n";
echo "<head><meta charset=\"UTF-8\"><title>PHP CGI</title></head>\n";
echo "<body style=\"font-family: sans-serif; background:#0b1020; color:#eaf0ff; padding:40px;\">";
echo "<h1>PHP CGI works</h1>";
echo "<p>Generated at " . date('c') . "</p>";
echo "<p>Request URI: " . htmlspecialchars($_SERVER['REQUEST_URI'] ?? '', ENT_QUOTES, 'UTF-8') . "</p>";
echo "</body></html>";

File: www/cgi-bin/hello.py
#!/usr/bin/env python3
import datetime
import os

print("Content-Type: text/html; charset=UTF-8")
print()
print("<!DOCTYPE html>")
print("<html lang='en'>")
print("<head><meta charset='UTF-8'><title>Python CGI</title></head>")
print("<body style='font-family: sans-serif; background:#0b1020; color:#eaf0ff; padding:40px;'>")
print("<h1>Python CGI works</h1>")
print(f"<p>Generated at {datetime.datetime.utcnow().isoformat()}Z</p>")
print(f"<p>Query string: {os.environ.get('QUERY_STRING', '')}</p>")
print("</body></html>")